apps/interop-rxjs-signal/src/app/list/photos.component.ts

```diff
@@ -80,40 +91,88 @@ import { PhotoStore } from './photos.store';
       </section>
     </ng-container>
   `,
-  providers: [provideComponentStore(PhotoStore)],
   host: {
     class: 'p-5 block',
   },
 })
-export default class PhotosComponent implements OnInit {
-  store = inject(PhotoStore);
-  readonly vm$ = this.store.vm$.pipe(
-    tap(({ search }) => {
-      if (!this.formInit) {
-        this.search.setValue(search);
-        this.formInit = true;
-      }
-    })
+export default class PhotosComponent extends SignalState<PhotoComponentState> {
```

Comment:
When do you decide you should create a "local" service instead of using inheritance ?

Reply:
In a component, I use inheritance for the state. If it would become so big that we need to split it up (composition) then I would extract the state as well. I just like to treat components as state machines